caddyhttp/proxy/upstream.go

```diff
@@ -371,7 +382,16 @@ func (u *staticUpstream) healthCheck() {
 	for _, host := range u.Hosts {
 		hostURL := host.Name + u.HealthCheck.Path
 		var unhealthy bool
-		if r, err := u.HealthCheck.Client.Get(hostURL); err == nil {
+
+		// set up request, needed to be able to modify headers
+		req, _ := http.NewRequest("GET", hostURL, nil)
```

Comment:
Check the error - if the URL is bad (thus resulting in an error), the upstream is considered unhealthy.